### ISSUE TITLE
password[retrieval_step] option is ignored when set to on/true/1/'1'

### ISSUE THREAD

**Issue by tberta** [pglombardo/PasswordPusher#1838]
## 🐛 Bug Report

I can't get the `password[retrieval_step]` option to work, no matter what the value is enabled with: `on`/`true`/`1`/`'1'`. 
The server responds with an JSON indicating that the option is : `true`
But when I use the generated URL, the option is not honoured.

## 🔬 How To Reproduce

Steps to reproduce the behaviour:

```
curl -X POST --data "password[payload]=mypassword&password[expire_after_days]=2&password[expire_after_views]=10&password[retrieval_step]=on" https://pwpush.com/p.json
```
Server replies that the retrieval_step option is `true`, but this option is not honoured when I test the url_token.
Same result if I use the API with authentication.
```
{
  "expire_after_days": 2,
  "expire_after_views": 10,
  "expired": false,
  "url_token": "w0jh2ekk2ssuo-at",
  "created_at": "2024-01-29T17:36:31.967Z",
  "updated_at": "2024-01-29T17:36:31.967Z",
  "deleted": false,
  "deletable_by_viewer": true,
  "retrieval_step": true,
  "expired_on": null,
  "days_remaining": 2,
  "views_remaining": 10
}
```

### Environment

Where are you running/using Password Pusher?

- [X] pwpush.com

## 📈 Expected behaviour

To avoid spoiling the secret link when sharing in chat or link preview services, I expect the intermediate link page when this option is set to true.

## Side note : API page states incorrect / non-working values

Besides, API states that the correct values for this option `password[retrieval_step]` are: _Must be one of: true, false, 1, 0._

value 1 or '1' doesn't work in my test:
```
 curl -X POST --data "password[payload]=mypassword&password[expire_after_days]=2&password[expire_after_views]=10&password[retrieval_step]=1" https://pwpush.com/p.json 
```
```
 curl -X POST --data "password[payload]=mypassword&password[expire_after_days]=2&password[expire_after_views]=10&password[retrieval_step]='1'" https://pwpush.com/p.json 
```

Reply parsed with jq,:
```
{
  "expire_after_days": 2,
  "expire_after_views": 10,
  "expired": false,
  "url_token": "hnc3ru0djlmw",
  "created_at": "2024-01-29T17:33:23.227Z",
  "updated_at": "2024-01-29T17:33:23.227Z",
  "deleted": false,
  "deletable_by_viewer": true,
  "retrieval_step": false, <------ not expected
  "expired_on": null,
  "days_remaining": 2,
  "views_remaining": 10
}
```

Values `on` and `true` makes the server replies with the expected (but non working) option :  `"retrieval_step": true`

**Comment by pglombardo** [comment]
Good find on the API documentation - I'll fix that up.  The value check done is [here](https://github.com/pglombardo/PasswordPusher/blob/master/app/controllers/passwords_controller.rb#L469).

Re: `retrieval_step` - this option simply appends `/r` to the generated URL that is shown in the preview page.  In fact, you can take any generated secret URL, append `/r` and you will get the 1-click preliminary page first.

So even though this URL (https://pwpush.com/en/p/dlupfqpq4n8vqa) was generated _without_ the `retrieval_step` option, you can still append `/r` and get the 1-click through option:

https://pwpush.com/en/p/dlupfqpq4n8vqa/r

`retrieval_step` is really just a preference for URL generation but it's not enforced (like passphrase lockdown).  The option was designed for the UI that just got passed down to the JSON API.

Did I explain that well enough that it makes sense?

**Comment by pglombardo** [comment]
Ok excellent then.  The API docs will be fixed in the next release from #1840.  I'll close out this issue but if anything remains, let me know!

**Comment by pglombardo** [comment]
Hi @tberta  - just an update that the API has some changes coming soon that shouldn't break anything but just in case, see #3068.

Hope all is well!
